GitUI/UserControls/InteractiveGitActionControl.cs

```diff
@@ -0,0 +1,240 @@
+﻿using System;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using GitCommands;
+using Microsoft.VisualStudio.Threading;
+using ResourceManager;
+
+namespace GitUI.UserControls
+{
+    public partial class InteractiveGitActionControl : GitModuleControl
+    {
+        private readonly TranslationString _progressMessage = new TranslationString("{0} is currently in progress.");
+        private readonly TranslationString _progressWithConflictsMessage = new TranslationString("{0} is currently in progress with merge conflicts.");
+
+        private readonly TranslationString _bisect = new TranslationString("Bisect");
+        private readonly TranslationString _rebase = new TranslationString("Rebase");
+        private readonly TranslationString _merge = new TranslationString("Merge");
+        private readonly TranslationString _patch = new TranslationString("Patch");
+
+        public enum State
+        {
+            None,
+            Bisect,
+            Rebase,
+            Merge,
+            Patch
+        }
+
+        private State _state = State.None;
+        private bool _hasConflicts = false;
+
+        public event EventHandler BisectEvent;
```

Comment:
Why is it `public`? How it is expected to be used?

Reply:
I couldn't work out how to open FormBisect from within the control, so this event tells FormBrowse to open it.

Reply:
IIUIC, you delegate it to accommodate this:
```cs
        private void BisectClick(object sender, EventArgs e)
        {
            using (var frm = new FormBisect(RevisionGrid))
            {
                frm.ShowDialog(this);
            }

            UICommands.RepoChangedNotifier.Notify();
        }
```

Is there a reason we can't do this in this module?
```cs
        private void BisectClick(object sender, EventArgs e)
        {
            if (!(FindForm() is FormBrowse form))
            {
                return;
            }

            using (var frm = new FormBisect(form.RevisionGridControl))
            {
                frm.ShowDialog(this);
            }

            form.UICommands.RepoChangedNotifier.Notify();
        }
```